### ISSUE TITLE
Obsolete attributes are order-dependant

### ISSUE THREAD

**Issue by alrz** [dotnet/roslyn#23824]
**Version Used**: https://github.com/dotnet/roslyn/commit/37ef0cdea72622fa9195b770808624c8dca007b3

**Steps to Reproduce**:

```cs
[Obsolete]
[Experimental]
class A
{
}

[Deprecated(null, DeprecationType.Deprecate, 0)]
[Experimental]
class B
{
}

class C
{
    static A F() => null; // warning CS8305: 'A' is for evaluation purposes only
    static B G() => null; // warning CS8305: 'B' is for evaluation purposes only
}
```

```cs
[Experimental]
[Obsolete]
class A
{
}

[Experimental]
[Deprecated(null, DeprecationType.Deprecate, 0)]
class B
{
}

class C
{
    static A F() => null; // warning CS0612: 'A' is obsolete
    static B G() => null; // warning CS0612: 'B' is obsolete
}
```

**Expected Behavior**: Obsolete should not be order-dependant.

**Actual Behavior**: Obsolete attributes are order-dependant.

**Comment by jcouv** [comment]
This does look like a bug.

The feature [spec](https://github.com/dotnet/roslyn/blob/master/docs/features/ExperimentalAttribute.md) doesn't mention the order, only that:
> Warnings and errors for [Obsolete] and [Deprecated] are reported instead of [Experimental] if there are multiple attributes.

**Comment by alrz** [comment]
The bug is where we overwrite `ObsoleteAttributeData` property on each `EarlyDecodeDeprecatedOrExperimentalOrObsoleteAttribute` usage. I wonder how we could keep track of the previous `ObsoleteAttributeData` since the `ObsoleteAttributeData` property can't be changed once read.

**Comment by gafter** [comment]
> Warnings and errors for [Obsolete] and [Deprecated] are reported instead of [Experimental] if there are multiple attributes.

Unfortunately, that's not what happens.  We respect the last one instead of the documented priority, as demonstrated by OP's first code example.
